### PR TITLE
++ Multibox Support

### PR DESCRIPTION
**Pull request by derekShaheen**
This change allows multi-box support by attempting to select the active D2R window and pulling data from it. Supports any number of active clients.

### CODE REVIEW

**Comment by derekShaheen** [comment]
Gif of change in action
https://media.giphy.com/media/4wSad7lQCfRgh1nQGk/giphy.gif

**Comment by OneXDeveloper** [comment]
We should likely still try to find processes by name and then just see if we have any that are the current foreground window. That way we don't randomly hook up to some other foreground process.

**Comment by derekShaheen** [comment]
That's a good callout, I'll implement that check.

File: Helpers/GameMemory.cs [hunk 1]
```diff
@@ -26,35 +26,56 @@
...
-        private static string processName = Encoding.UTF8.GetString(new byte[] { 68, 50, 82 });
-        public static IntPtr? ProcessHandle = null;
-        public static bool foundcheck = false;
-        public static IntPtr? SaveAddress = null;
-        public static IntPtr? CheckAddress = null;
+        private static string ProcessName = Encoding.UTF8.GetString(new byte[] { 68, 50, 82 });
+        private static IntPtr AdrPlayerUnit = IntPtr.Zero;
+        private static IntPtr PtrPlayerUnit = IntPtr.Zero;
+
+        [DllImport("user32.dll")]
```
Comment (OneXDeveloper) [inline Helpers/GameMemory.cs:39]: I believe these are already imported in the Windows External class.

**Comment by OneXDeveloper** [comment]
Looks like you’ll also need to rebase the branch to pull in changes.

**Comment by AudriusButkevicius** [comment]
Tbh, I feel this should be different. We should check for d2r on startup, if there is one, do the usual, if there are multiple, pop a dialog asking which one to pick. The whole "its the one that is active" is magical and non-obvious to users.

**Comment by derekShaheen** [comment]
Thanks OneX, I'll have to figure out which branch to rebase to, lol.

And AudriusButkevicius, I began implementing that initially, but I found that it would be considerably less efficient. I don't see a reason to open this program multiple times though (one for each client), when it can be easily configured to "just work" for whatever client you have open. Let it be a little magical, it just works, ha.

**Comment by AudriusButkevicius** [comment]
Also, the current code lists active window AND every process on the machine on every render loop, and doesn't cache anything.

**Comment by derekShaheen** [comment]
Those are fair points of criticism, however the current main implementation pulls every process without caching every frame as well. I simply added the request about the foreground window, in order to compare it to the process list that was already acquired, in order to determine which process to attach to. The performance analysis showed a negligible difference after my change.

File: Helpers/GameMemory.cs (hunk 1 above)
Comment (derekShaheen) [reply Helpers/GameMemory.cs]: Resolved with further commit. Reference removed / moved to windows external class.

**Comment by derekShaheen** [comment]
Thank you, OneX!

**Comment by OneXDeveloper** [comment]
#120 is similar idea. I’ve also got an isboxer setup now and can poke at this a bit later this week.

**Comment by misterokaygo** [comment]
#123 will be addressing this, thank your for your support.

**Pull request closed without merging**